pkg/web/scene/scene.go

```diff
@@ -73,15 +92,48 @@ func (s Scene) GetUser() *auth.Claims {
 	return nil
 }
 
-func (s Scene) IsAuthenticated() bool {
-	if isauths, ok := s[IsAuthenticated]; ok {
-		return isauths.(bool)
+func (s Scene) HasRole(role string) bool {
+	if user := s.GetUser(); user != nil {
+		return user.Role == role
 	}
 	return false
 }
 
-func GetAuthUserRole(c *gin.Context) string {
-	ctx := New(c)
-	user := ctx.GetUser()
-	return user.Role
+func (s Scene) IsAdmin() bool {
+	return s.HasRole(RoleAdmin)
+}
+
+func (s Scene) IsViewOnly() bool {
+	return s.HasRole(RoleObserver)
+}
+
+//===========================================================================
+// Scene API Data Related Helpers
+//===========================================================================
+
+func (s Scene) AccountsList() *api.AccountsList {
+	if data, ok := s[APIData]; ok {
+		if out, ok := data.(*api.AccountsList); ok {
+			return out
+		}
+	}
+	return nil
+}
+
+func (s Scene) UserList() *api.UserList {
+	if data, ok := s[APIData]; ok {
+		if out, ok := data.(*api.UserList); ok {
+			return out
+		}
+	}
+	return nil
+}
+
+func (s Scene) CounterpartyList() *api.CounterpartyList {
+	if data, ok := s[APIData]; ok {
+		if out, ok := data.(*api.CounterpartyList); ok {
+			return out
+		}
+	}
+	return nil
```

Comment:
Unfortunately these are now required in order to get the api data out of scene into the template using a `{{ with }}` block.